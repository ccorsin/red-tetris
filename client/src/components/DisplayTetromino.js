import React from 'react';

import { StyledDisplay, StyledDisplayCase } from './styles/StyledDisplay';
import { StyledTetromino, StyledRow } from './styles/StyledTetromino';
import Cell from './Cell';

import { TETROMINOS } from "../utils/tetrominos-colors.js";

const color = (cell) => {
  return TETROMINOS[cell].color;
}

const Display = ({ text, tetro }) => (
  <StyledDisplay>
    <StyledDisplayCase>
      <p>{text}</p>
      {tetro && tetro.shape ?
        <StyledTetromino width={tetro.shape[0].length} height={tetro.shape.length}>
          {tetro.shape.map((row, y) => (
            <StyledRow className="row" key={y}>
              {row.map((cell, x) => <Cell key={x} type={cell[0]} color={color(cell)} size={1.5} cell={true} />)}
            </StyledRow>
          ))}
        </StyledTetromino> :
        <div></div>
      }

    </StyledDisplayCase>
  </StyledDisplay >
);

export default Display;
