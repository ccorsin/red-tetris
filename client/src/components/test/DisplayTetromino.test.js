import React from 'react'
import { shallow } from 'enzyme';
import DisplayTetromino from '../Display';
import toJson from "enzyme-to-json"

describe('<DisplayTetromino/> Component', () => {
  it('rendering correctly DisplayTetromino', () => {
    let wrapper = shallow(
      <DisplayTetromino text='text' tetro='L'/>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})