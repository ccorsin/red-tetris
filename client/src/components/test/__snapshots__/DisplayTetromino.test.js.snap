// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayTetromino/> Component rendering correctly DisplayTetromino 1`] = `
<styled.div>
  <styled.div>
    <p>
      text
    </p>
    <styled.div
      height={2}
      width={2}
    >
      <styled.div
        className="row"
        key="0"
      >
        <Memo(Cell)
          cell={true}
          color="255, 255, 0"
          key="0"
          size={1.5}
          type="O"
        />
        <Memo(Cell)
          cell={true}
          color="0, 0, 0"
          key="1"
          size={1.5}
          type="0"
        />
      </styled.div>
      <styled.div
        className="row"
        key="1"
      >
        <Memo(Cell)
          cell={true}
          color="255, 255, 0"
          key="0"
          size={1.5}
          type="O"
        />
        <Memo(Cell)
          cell={true}
          color="255, 255, 0"
          key="1"
          size={1.5}
          type="O"
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
